lib/game/client/game-engine.js

```diff
@@ -6,93 +6,163 @@
  *
  */
 
-const { Game, Input } = require('phaser');
 const TemplateEngine = require('mustache');
 const { GameConst } = require('../constants');
 const { ObjectsConst } = require('../../objects/constants');
 const { sc } = require('@reldens/utils');
+const { UiSceneManager } = require("./ui-scene-manager");
 
-class GameEngine extends Game
+class GameEngine
 {
 
     constructor(props)
     {
-        super(props.config);
         // @TODO - BETA - Refactor the entire class:
         //      - Extract all Phaser methods into the engine driver class and implement the engine on the GameManager.
         //      - Extract the template parsing into a new "template" or "elements" domain driver.
         //      - Extract the screen resize methods into a new ScreenHandler class.
         //      - Extract the tab target methods into a new TabPlugin class.
+
+        // @TODO: remove this manager from here. Circular references between objects is a bad practice.
```

Comment:
everything can be controlled from everywhere, the gameManager encapsulte the entire thing, so we will always have circurlar references unless we refactor the thing from scratch and use let's say the global context to avoid injecting the other objects in the ones we need (which at the end will be everywhere :P ), let's not do that for now at least :D

Reply:
Is hard to understand where is the limit and try to get arround the concepts used in web development, but in that case why just pass gameManager everywhere as a parameter (or event better, access it by a global scope), and then access what is necessary from there. I understand it is more of an architecture/desing code discussion and that is long and tedious and that is not a priority right now, but the logic of everything is access from everywhere won't hurt the game functionality but it will make it harder to maintain and scale the code, imo.